dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,325 @@
+        #  seed = hash(((k, v) for k, v in meta.items())) % 2**32
+        #  rng = np.random.RandomState(seed)
+        #  score = rng.rand()
+    else:
+        # TODO: change this to dask_ml.get_scorer
+        score = model.score(x, y)
+    meta.update(score=score)
+    assert meta['iterations'] > 0
+    meta['mean_score_time'] += time() - start
+    meta['mean_test_score'] = score
+    return meta
+
+
+def _top_k(results, k=1, sort="score"):
+    res = toolz.topk(k, results, key=sort)
+    return list(res)
+
+
+def _to_promote(result, completed_jobs, eta=None, asynchronous=None):
+    bracket_models = [r for r in completed_jobs
+                      if r['bracket_iter'] == result['bracket_iter'] and
+                      r['bracket'] == result['bracket']]
+
+    if not asynchronous:
+        bracket_completed = len(bracket_models) == result['num_models']
+        if bracket_completed:
+            to_keep = len(bracket_models) // eta
+            if to_keep <= 1:
+                return []
+            top_results = _top_k(bracket_models, k=to_keep, sort="score")
+            for job in top_results:
+                job["num_models"] = to_keep
+                job['partial_fit_calls'] *= eta
+                job['bracket_iter'] += 1
+            return top_results
+        return []
+    raise ValueError
+
+
+def _create_model(model, params, random_state=42):
+    model = clone(model).set_params(**params)
+    if 'random_state' in model.get_params():
+        model.set_params(random_state=random_state)
+    #  meta = {}  # right now no meta information
+    return model, None
+
+
+def _bracket_name(s):
+    return "bracket={s}".format(s=s)
+
+
+def _model_id(s, n_i):
+    return "bracket={s}-{n_i}".format(s=s, n_i=n_i)
+
+
+async def _hyperband(model, params, X, y, max_iter=None, eta=None,
+                     dry_run=False, fit_params={}, random_state=42,
+                     asynchronous=None):
+    client = default_client()
+    rng = check_random_state(random_state)
+    N, R, brackets = _get_nr(max_iter, eta=eta)
+    params = iter(ParameterSampler(params, n_iter=sum(N),
+                                   random_state=random_state))
+    if isinstance(model, Incremental):  # this should probably be handled elsewhere
+        model = model.estimator
+    params = {_model_id(s, n_i): next(params)
+              for s, n, r in zip(brackets, N, R) for n_i in range(n)}
+    model_futures = {_model_id(s, n_i):
+                     client.submit(_create_model, model,
+                                   params[_model_id(s, n_i)],
+                                   random_state=random_state)
+                     for s, n, r in zip(brackets, N, R) for n_i in range(n)}
+
+    # lets assume everything in fit_params is small and make it concrete
+    fit_params = await client.compute(fit_params)
+
+    r = train_test_split(X, y, test_size=0.15, random_state=random_state)
```

Comment:
What does Scikit-Learn do in cases like this?  Do they hardcode things or is there a convention behind a particular parameter name that we can use?